### PR TITLE
[12.0][IMP] agreement_serviceprofile: add stage

### PR DESCRIPTION
**Pull request by brian10048**
- Add stage_type to agreement.stage
- Track stage of service profile
- Add Kanban view for service profiles to the dashboard
- Add product_id field to service profile

@max3903 for #349

### CODE REVIEW

**Comment by brian10048** [comment]
Please excuse the mess! 

@max3903, should I re-use the agreement.stage and add a selection type field for service profile stages like was done with field service stages?

**Comment by max3903** [comment]
Yes please.

**Comment by max3903** [comment]
@brian10048 I added a product_id field on the design with a domain. Can you implement it please?

**Review by max3903 — APPROVED** [review]

**Comment by max3903** [comment]
Ping @kvedder-amplex @barnarj

**Comment by brian10048** [comment]
@max3903 I reworked the stage to use the agreement.stage model

> 
> @brian10048 I added a product_id field on the design with a domain. Can you implement it please?

I'm trying to figure out what you mean on the design.  Could you explain a little more maybe?

**Comment by max3903** [comment]
@brian10048 The service profile is the same to the service product as the serial number to the stockable product.

Screenshot from 2019-07-12 13-51-07

So with a product_id field on the service profile, we can access all the service information defined on the product template.

**Comment by brian10048** [comment]
Thank you @max3903. That helps

Also, not sure what I am missing.  The dashboard kanban view for service profiles is not showing the stages unless there is a record for that stage

image

File: agreement_serviceprofile/models/agreement_serviceprofile.py [hunk 1]
```diff
@@ -2,14 +2,19 @@
...
 class AgreementServiceProfile(models.Model):
     _name = 'agreement.serviceprofile'
     _description = 'Agreement Service Profiles'
 
+    def _default_stage_id(self):
+        return self.env.ref('agreement_serviceprofile.servpro_stage_draft')
+
     name = fields.Char(string="Name", required=True)
+    stage_id = fields.Many2one('agreement.stage', string="Stage",
+                               default=_default_stage_id, copy=False)
```
Comment (max3903) [inline agreement_serviceprofile/models/agreement_serviceprofile.py]:
```suggestion
                               default=_default_stage_id, copy=False,
                               group_expand='_read_group_stage_ids',)
```

Comment (max3903) [reply agreement_serviceprofile/models/agreement_serviceprofile.py]: @brian10048 That should open all the stages, even without records.

Comment (brian10048) [reply agreement_serviceprofile/models/agreement_serviceprofile.py]: Thank you.  I added the method, but forgot that part in the field definition!

**Comment by brian10048** [comment]
@max3903 I believe this is done now.  Cleaned up into one commit.  Let me know if there is anything else I need to do.

**Comment by max3903** [comment]
:+1: LGTM. Tested on Runbot.

**Review by bodedra — APPROVED** [review]
LTGM :+1: Good work @brian10048 

Not blocking comment: IMO for better maintain and commit history, shouldn't we have two PRs for these ? one for `agreement_legal` and another for `agreement_serviceprofile`

**Pull request closed without merging**